rust/datafusion/src/datasource/parquet.rs

```diff
@@ -65,6 +66,7 @@ impl TableProvider for ParquetTable {
         &self,
         projection: &Option<Vec<usize>>,
         batch_size: usize,
+        _filters: &[Expr],
```

Comment:
I honestly know very little about the internals of Parquet and I'd like to avoid too much extra complexity in this PR, I wonder if making use of this in the various existing implementations where feasible would work better as a followup task (or even task per provider)?

Reply:
@returnString it's great to see someone else working on predicate push-down as well; 
I have been working on this for a couple of weeks, targeting an end-to-end implementation for parquet and have done similar changes to the filter push-down optimizer but your implementation is better because of the idea for full vs partial filter push-down; in my version I have `predicate: &Option<Expr>`, but `filters: &[Expr]` should work as well;

I think it makes sense to separate the generic support for predicate push-down to the data source from the implementation for various data sources such as parquet because each change will be fairly big so makes sense to split into smaller changes;

regarding a parquet implementation of predicate push-down I have been working on the idea of building arrays from the min / max statistics in row groups and then reusing the existing physical expressions already implemented in datafusion; I already have the code that builds statistics arrays, next working on the expression evaluation - hopefully will have enough to start a PR in the next couple of weeks;

Reply:
@yordan-pavlov Likewise!

Are you broadly happy that this interface and implementation would suffice for your Parquet-specific work? My use case is an entirely external TableProvider implementation, so hopefully that's kept my assumptions to a minimum.

If you've got time, seeing as you've been working in this area, I'd really appreciate any other thoughts you might have here :)

Reply:
> I think it makes sense to separate the generic support for predicate push-down to the data source from the implementation for various data sources such as parquet because each change will be fairly big so makes sense to split into smaller changes;

I think this approach makes a lot of sense. 

I have also been thinking about pushing predicates down into custom table providers, and FWIW I think this interface would work well for my use.

Reply:
Okay for doing it later, I was just curious to confront the API changes with a concrete use case. If Yordan tried it out, I'm reassured 😉